### PR TITLE
Refactor irq

### PR DESCRIPTION
**Pull request by code-mm**
Changes
- Changes naming of registers to match ARM documentation.
- Fixes bug that not all 10 Bit of the `GICC_IAR` register is used.
- Changes checked interrupt for Mini UART to the OR-ed `AUX` interrupt on the VideoCore.
- Removes invalid interrupt activation.
- Refactors the `handle_irq` function to use a switch-case statement.
- Adds comments and changes source formatting.

### CODE REVIEW

**Comment by code-mm** [comment]
What are you using to format the source code?

**Comment by anarchuser** [comment]
[ClangFormat](https://clang.llvm.org/docs/ClangFormat.html)
The .clang-format is in the root directory of this project

**Review by anarchuser — DISMISSED** [review]
Looks good.
I assume this builds and runs as before.

Two unimportant things to take note of:

File: thorn/src/kernel/irq.c [hunk 1]
```diff
@@ -25,28 +25,28 @@ void show_invalid_entry_message (int type, ptr_t esr, ptr_t address) {
...
 }
 
 void handle_irq (void) {
-    unsigned int irq_ack_reg = get32 (GICC_IAR);
-    unsigned int irq1        = irq_ack_reg & 0x2FF;
-    if (irq1 & SYSTEM_TIMER_IRQ_1) {
-        put32 (GICC_EOIR, irq_ack_reg);
-        handle_timer_irq ();
-    }
+    unsigned int irq = get32 (GICC_IAR) & 0x3FF; // Use first 10 Bit of register
```
[the remark this answers is not in the document]
Comment (code-mm) [reply thorn/src/kernel/irq.c]: Got I idea. Just changed...

**Review by anarchuser — APPROVED** [review]

File: thorn/src/kernel/irq.c [hunk 2]
```diff
@@ -25,28 +25,28 @@ void show_invalid_entry_message (int type, ptr_t esr, ptr_t address) {
...
     assign_target (SYSTEM_TIMER_IRQ_1, 0);
-    put32 (ENABLE_IRQS_1, SYSTEM_TIMER_IRQ_1
-#ifdef _ROSE_K_MINI_UART_H
-                                  | ENABLE_AUX_INT
-#endif
-    );
+    assign_target (AUX_CUMULATIVE, 0);
+
     enable_interrupt (SYSTEM_TIMER_IRQ_1);
+    enable_interrupt (AUX_CUMULATIVE);
```
[the remark this answers is not in the document]
Comment (code-mm) [reply thorn/src/kernel/irq.c:34]: Just added the `ifdef` again. Missed that when refactoring. Thanks!

**Pull request merged**